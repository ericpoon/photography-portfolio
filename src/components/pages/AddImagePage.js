import React from 'react';
import { connect } from 'react-redux';
import { startAddImage } from '../../actions/images';
import ImageCard from '../ImageCard';

export const AddImagePage = (props) => {
  return (
    <div className={'content-container content-container--small'}>
      <ImageCard
        editable
        onSaveClick={updates => props.startAddImage(updates)}
      />
    </div>
  );
};

export default connect(undefined, { startAddImage })(AddImagePage);
