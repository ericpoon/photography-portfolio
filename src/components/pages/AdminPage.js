import React from 'react';
import { connect } from 'react-redux';
import { startEditImage, startDeleteImage } from '../../actions/images';
import ImageCardCollection from '../ImageCardCollection';
import ImageCard from '../ImageCard';

export const AdminPage = (props) => {
  return (
    <div>
      <div className={'content-container'}>
        <ImageCardCollection>
          {props.images.map((image) => {
            return (
              <ImageCard
                {...image}
                editable
                key={image.id}
                onSaveClick={updates => props.startEditImage(image.id, updates)}
                onDeleteClick={() => props.startDeleteImage(image.id)}
              />);
          })}
        </ImageCardCollection>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  images: state.images,
});

export default connect(mapStateToProps, { startEditImage, startDeleteImage })(AdminPage);
