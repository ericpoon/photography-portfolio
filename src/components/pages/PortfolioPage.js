import React from 'react';
import { connect } from 'react-redux';
import { showDetail } from '../../actions/gallery';
import ImageDetailModal from '../ImageDetailModal';
import ImageBoard from '../ImageBoard';
import ImageCard from '../presentational/ImageCard';

export const PortfolioPage = props => (
  <div>
    <ImageDetailModal />
    <div className={'content-container'}>
      <ImageBoard>
        {props.images.map((image) => {
          return (
            <ImageCard
              {...image}
              key={image.id}
              onImageClick={() => props.showDetail(image.id)}
            />
          );
        })}
      </ImageBoard>
    </div>
  </div>
);

const mapStateToProps = state => ({
  images: state.images,
});

export default connect(mapStateToProps, { showDetail })(PortfolioPage);
