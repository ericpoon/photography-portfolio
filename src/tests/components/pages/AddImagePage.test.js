import React from 'react';
import { shallow } from 'enzyme';
import { AddImagePage } from '../../../components/pages/AddImagePage';

let wrapper,
  startAddImage;

beforeEach(() => {
  startAddImage = jest.fn();
  wrapper = shallow(
    <AddImagePage
      startAddImage={startAddImage}
    />);
});

it('renders AddImagePage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('handles addImage correctly', () => {
  const imageToAdd = { title: 'title', subtitle: 'subtitle' };
  wrapper.find('EditableImageCard').prop('onSaveClick')(imageToAdd);
  expect(startAddImage).toHaveBeenCalledWith(imageToAdd);
});
