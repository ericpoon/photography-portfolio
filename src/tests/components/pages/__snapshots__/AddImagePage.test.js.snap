// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddImagePage correctly 1`] = `
<div
  className="content-container content-container--small"
>
  <ImageCard
    editable={true}
    onSaveClick={[Function]}
  />
</div>
`;
